src/main/java/org/spongepowered/common/mixin/core/item/inventory/MixinContainerHorseChest.java

```diff
@@ -0,0 +1,16 @@
+package org.spongepowered.common.mixin.core.item.inventory;
+
+import net.minecraft.inventory.ContainerHorseChest;
+import org.spongepowered.api.item.inventory.type.CarriedInventory;
+import org.spongepowered.asm.mixin.Mixin;
+
+import java.util.Optional;
+
+@Mixin(ContainerHorseChest.class)
+public abstract class MixinContainerHorseChest implements CarriedInventory {
+
+    @Override
+    public Optional getCarrier() {
+        return Optional.empty(); // TODO can we get the horse in here somehow?
```

Comment:
Could, technically, throw a mixin into `net.minecraft.entity.passive.AbstractHorse#initHorseChest` to set the carrier?